### PR TITLE
Resource tests

### PR DESCRIPTION
**Pull request by ilholmes**
Add tests for resources.py classes. Still need to test PanoptesResourceCache.

### CODE REVIEW

File: tests/test_framework.py [hunk 1]
```diff
@@ -164,12 +228,134 @@ def test_panoptes_resource_set(self):
...
+
+            # Test key not found
+            mock_find_keys = Mock(
+                return_value=['dummy',
+                              'plugin|test|site|test|class|test|subclass|test|type|test|id|test|endpoint|test'])
+            with patch('yahoo_panoptes.framework.resources.PanoptesKeyValueStore.find_keys',
+                       mock_find_keys):
+                self.assertEqual(1, len(panoptes_resource_store.get_resources()))
+
+            # Test resource store methods raise correct errors
```
Comment (varunvarma) [inline tests/test_framework.py:334]: Need to test empty and non-string parameter to get_resource

File: tests/test_framework.py [hunk 2]
```diff
@@ -164,12 +228,134 @@ def test_panoptes_resource_set(self):
...
+            mock_get = Mock(side_effect=Exception)
+            with patch('yahoo_panoptes.framework.resources.PanoptesKeyValueStore.get', mock_get):
+                with self.assertRaises(Exception):
```
Comment (varunvarma) [inline tests/test_framework.py]: Could you please change get_resource to raise a PanoptesResourceError instead of a bare Exception in get_resource
